tests/test_physics/test_ansatz.py

```diff
@@ -355,3 +356,216 @@ def test_bargmann_Abc_to_phasespace_cov_means(self):
         assert math.allclose(new_state_means22[0], state_means)
         assert math.allclose(new_state_coeff1[0], 1.0)
         assert math.allclose(new_state_coeff22[0], 1.0)
+
+
+class TestDiffOpPolyExpAnsatz:
+    r"""
+    Tests the ``DiffOpPolyExpAnsatz`` class.
+    """
+
+    Abc_n1 = Abc_triple(1)
+    Abc_n2 = Abc_triple(2)
+    Abc_n3 = Abc_triple(3)
+
+    @pytest.mark.parametrize("triple", [Abc_n1, Abc_n2, Abc_n3])
+    def test_init(self, triple):
+        A, b, c = triple
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        assert np.allclose(ansatz.mat[0], A)
+        assert np.allclose(ansatz.vec[0], b)
+        assert np.allclose(ansatz.array[0], c)
+
+    def test_add(self):
```

Comment:
Adding them together didn't work if the c's have different lengths. This returned an error.

```
A1 = np.ones((1,2,2))
b1 = np.ones((1,2))
c1 = np.array([[1,2,3,4,5,6]])

A2 = np.ones((1,2,2))
b2 = np.ones((1,2))
c2 = np.array([[1,2,3,4,5]])

ansatze1 = DiffOpPolyExpAnsatz(A1,b1,c1)
ansatze2 = DiffOpPolyExpAnsatz(A2,b2,c2)

ansatze1 + ansatze2
```

Reply:
But would we expect that to work? One can also not add two ansatze together if the A and b are different.

Reply:
But would we expect this to work? We can also not add ansatze that have different size of A and b.